### DISCUSSION TITLE
Computing feature's dependencies via goal?

### DISCUSSION THREAD

**Discussion by lppedd** [eclipse-tycho/tycho#2251] [Ideas]
Eclipse offers a way to re-compute a feature's dependencies with a button click.  

![image](https://user-images.githubusercontent.com/19871649/225873975-850d35aa-f014-4f8f-ae6b-7894d51a901e.png)

This can become very tedious when dealing with a large number of features. 
Could Tycho interact with the PDE API and perform the same action in a Maven-first way?

**Comment by laeubi** [comment]
First question is, why do you want to do that? There is actually no need to perform this action as Tycho / P2 can compute this anyways, so usually you would only add items that are **not** automatically discoverable here.

**Reply by lppedd** [reply]
Mmm interesting. Then a follow up question would be, what's the point of the Compute action if it's done anyway at some point?

**Reply by laeubi** [reply]
Historic reasons, people like to press a button to "see" all dependencies, ... probably more :-)

Technically best is to leave ot completely empty and only add things you has choose because of a reason, e.g. it could be that your feature does only require another feature to be present but you are not directly depend on it.

**Reply by lppedd** [reply]
@laeubi I guess I've just discovered I've been pressing a button for no reason 😢 
Thanks!

https://lostpedia.fandom.com/wiki/Pushing_the_button

**Reply by laeubi** [reply]
It was even suggested at pde to remove the button but I'm not sure what was the status there...
